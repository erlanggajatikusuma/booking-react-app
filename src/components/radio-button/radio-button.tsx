import React, {useCallback} from 'react';
import {TouchableOpacity, Text, ViewStyle, TextStyle, View} from 'react-native';
import {color} from '../../theme';

interface RadioButtonProps {
  label: string;
  value: number;
  selected: boolean;
  onSelect: (value: number) => void;
  style?: ViewStyle;
  labelStyle?: TextStyle;
}

const CONTAINER: ViewStyle = {
  flexDirection: 'row',
  alignItems: 'center',
};

const RADIO: ViewStyle = {
  width: 20,
  height: 20,
  borderRadius: 10,
  borderWidth: 2,
  alignItems: 'center',
  justifyContent: 'center',
  backgroundColor: 'transparent',
};

const RADIO_INNER: ViewStyle = {
  width: 12,
  height: 12,
  borderRadius: 6,
  backgroundColor: color.darkBlue,
};

const LABEL: TextStyle = {
  marginLeft: 8,
  fontSize: 16,
};

export const RadioButton: React.FC<RadioButtonProps> = React.memo(
  ({label, value, selected, onSelect, style, labelStyle}) => {
    const handlePress = useCallback(() => {
      onSelect(value);
    }, [onSelect, value]);

    return (
      <TouchableOpacity onPress={handlePress} style={[CONTAINER, style]}>
        <View
          style={[
            RADIO,
            {
              borderColor: selected ? color.darkBlue : 'gray',
            },
          ]}>
          {selected && <View style={RADIO_INNER} />}
        </View>
        <Text style={[LABEL, labelStyle]}>{label}</Text>
      </TouchableOpacity>
    );
  },
);
