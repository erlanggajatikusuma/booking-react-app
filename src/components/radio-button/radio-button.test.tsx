import React from 'react';
import renderer from 'react-test-renderer';
import {RadioButton} from './radio-button';

describe('RadioButton Component', () => {
  it('renders correctly with label and unselected state', () => {
    const tree = renderer
      .create(
        <RadioButton
          label="Option 1"
          value={1}
          selected={false}
          onSelect={() => {}}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with label and selected state', () => {
    const tree = renderer
      .create(
        <RadioButton
          label="Option 2"
          value={2}
          selected={true}
          onSelect={() => {}}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls onSelect when pressed', () => {
    const onSelectMock = jest.fn();
    const tree = renderer.create(
      <RadioButton
        label="Option 3"
        value={3}
        selected={false}
        onSelect={onSelectMock}
      />,
    );

    // Find the radio button element by its testID
    const radioOuterElement = tree.root.findByProps({testID: 'radio-outer'});

    // Simulate a button press on the radio button
    radioOuterElement.props.onPress();

    expect(onSelectMock).toHaveBeenCalledWith(3);
  });
});
