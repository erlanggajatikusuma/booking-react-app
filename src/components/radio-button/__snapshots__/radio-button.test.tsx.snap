// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton Component renders correctly with label and selected state 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="radio-outer"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        {
          "borderColor": "#004492",
        },
      ]
    }
  >
    <View
      style={
        {
          "backgroundColor": "#004492",
          "borderRadius": 6,
          "height": 12,
          "width": 12,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "marginLeft": 8,
        },
        undefined,
      ]
    }
  >
    Option 2
  </Text>
</View>
`;

exports[`RadioButton Component renders correctly with label and unselected state 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="radio-outer"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        {
          "borderColor": "gray",
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "fontSize": 16,
          "marginLeft": 8,
        },
        undefined,
      ]
    }
  >
    Option 1
  </Text>
</View>
`;
